qqmlsortfilterproxymodel.cpp

```diff
@@ -350,10 +350,7 @@ bool QQmlSortFilterProxyModel::lessThan(const QModelIndex& source_left, const QM
 {
     if (m_completed) {
         if (!m_sortRoleName.isEmpty()) {
-            if (QSortFilterProxyModel::lessThan(source_left, source_right))
-                return m_ascendingSortOrder;
-            if (QSortFilterProxyModel::lessThan(source_right, source_left))
-                return !m_ascendingSortOrder;
+            return QSortFilterProxyModel::lessThan(source_left, source_right) == m_ascendingSortOrder;
```

Comment:
I'd even directly return the opposite here:
```suggestion
            return QSortFilterProxyModel::lessThan(source_right, source_left) != m_ascendingSortOrder;
```

Reply:
If my inverted boolean brain works correctly here, this should ensure equal items do not swap positions (ie stable sort)

Reply:
Yeah, as `QSortFilterProxyModel::lessThan` returns bool, both of those two if conditions may be false and no return value is evaluated at this point. So to keep stable sorting it should be changed imo.

Reply:
Actually it's much simpler than that. No inverted brain required 😄 

Updated